### PR TITLE
Reviewed Content

### PR DESCRIPTION
**Pull request by FlexYourBrain**
Reviewed all projects in repository  making sure they all work in latest defold release 1.9.3. Various small changes were made.

- 15-puzzle: The file '/builtins/fonts/system_font.font' could not be found.

**Fix: replaced missing font with always_on_top.font**

- hands_on: The file '/builtins/fonts/system_font.font' could not be found. game.script mouse input needs updated to work.

**Fix: replaced missing fonts with always_on_top.font, updated game.script input to use action_id hash("touch"). Set Physics settings gravity-y to -200.0 & turned on used fixed timestep. Updated the project to a more progressed version that matches the image of the project in the hands on section of the slides.**

- material: WARNING:RENDER: This interface for render.draw() is deprecated.

**Fix: set constants as table in render.draw(self.fsquad_pred, {constants = constants})**

- platformer: The file '/hero/hero.go' could not be loaded: Invalid embedded component 'spinemodel' of type 'spinemodel' in '/hero/hero.go'. 

```!!! spine project needs to be re-authored with new version of spine and latest version of extension (https://github.com/defold/extension-spine/releases). I cannot author a renewed version.```

- shooter_hot_reload: Opening main.collection causes editor error(java.lang.NullPointerException: Cannot invoke "java.lang.Number.doubleValue()" because "x" is null),  WARNING:RENDER: Function 'render.draw_debug2d' is deprecated.

**Fix: java error due to materials grid.material and stars.materials constants having non existing values, old material updated to latest formatting. Removed render.draw_debug2d() from render script. Adjusted render script draws.**

- sidescroller: **adjusted background z position(was out of frustum) and ship max_speed set higher.**

- threedee: WARNING:RENDER: Function 'render.draw_debug2d' is deprecated, materials and animation needed updated.

**Fix: replaced draw_debug2d() with draw_debug3d(). Added model material with vertex space world. created animationset file and applied to model for playback.**

- **README.md: Updated with details about the original purpose of the content. Please review for accuracy.**

### CODE REVIEW

File: hands_on/main/ball.script [hunk 1]
```diff
@@ -4,7 +4,7 @@ local score = 0
 function on_message(self, message_id, message, sender)
 	if message_id == hash("collision_response") and message.group == hash("pin") then
 		go.delete(message.other_id)
-		score = score + 1
+		score = score + 35
```
Comment (britzl) [inline hands_on/main/ball.script]: Why this change?

File: shooter_hot_reload/main/enemycontroller.script [hunk 2]
```diff
@@ -133,7 +133,7 @@ end
 function on_message(self, message_id, message, sender)
 	if message_id == hash("start") then
 		self.waiting = true
-		self.t = 0
+		self.t = 8.2
```
Comment (britzl) [inline shooter_hot_reload/main/enemycontroller.script]: Why?

**Review by britzl — CHANGES_REQUESTED** [review]
Thanks. I have questions about two changes which seemed unrelated to the fixes.

File: hands_on/main/ball.script (hunk 1 above)
Comment (FlexYourBrain) [reply hands_on/main/ball.script]: I made the score a higher number while adjusting the label component to make sure there was enough room while the score incremented during playtime. Good catch I should have reverted back to 1 afterward.

File: shooter_hot_reload/main/enemycontroller.script (hunk 2 above)
Comment (FlexYourBrain) [reply shooter_hot_reload/main/enemycontroller.script]: during testing the value was made higher to spawn enemy waves faster. I should have reverted the change afterward.

**Comment by FlexYourBrain** [comment]
alright , unrelated testing changes have been reverted.

**Review by britzl — APPROVED** [review]
Thanks!
